support-workers/src/main/scala/com/gu/zuora/ZuoraService.scala

```diff
@@ -88,6 +89,22 @@ class ZuoraService(val config: ZuoraConfig, client: FutureHttpClient, baseUrl: O
     } yield ddId).value
   }
 
+  def getSubscriptionFromRedemptionCode(redemptionCode: RedemptionCode): Future[SubscriptionRedemptionQueryResponse] = {
+    val queryData = QueryData(
+      //TODO RB expire codes
+      s"select id, RedemptionCode__c, GifteeIdentityId__c from subscription where RedemptionCode__c = '${redemptionCode.value}' and status = 'Active'"
```

Comment:
generally it would be risky to insert some user input into a query directly, but since RedemptionCode is validated when it's wrapped I think we're ok as no quotes are possible.

Reply:
do we need `RedemptionCode__c ` at all in the select?

Reply:
Re the injection point, also worth noting that this is Zoql not SQL so it's a lot more constrained in what it can do.

On whether we need `RedemptionCode__c` in the select, we do check it in an integration test although we don't use it anywhere else. I think I'll leave it in unless you feel strongly that it should be removed.

Reply:
if it's not used then take it out as dependencies between us and zuora data are more expensive than compile checked ones within the code.  It's easier to remove it now when it's fresh than later when it's been in PROD a while.

Reply:
and yes Zoql is much more constrained so the risk is definitely much lower, and DS are not a big target, also it would be difficult to extract any PII via any issue with this query.
I only mentioned because this is the critical point where a vuln would be caused, when data is merged into a higher level structure, and we are relying on validation that happens elsewhere rather than proper escaping at the point of constructing the query.